### PR TITLE
Add useragent override for RN

### PR DESCRIPTION
**Pull request by rclee**

### CODE REVIEW

File: MapboxMobileEvents/MMEAPIClient.m [hunk 1]
```diff
@@ -100,6 +100,11 @@ - (NSURLRequest *)requestForEvents:(NSArray *)events {
 }
 
 - (void)setupUserAgent {
+    if ([self.applicationBundle objectForInfoDictionaryKey:@"MMEMapboxUserAgentBase"] && [self.applicationBundle objectForInfoDictionaryKey:@"MMEMapboxHostSDKVersion"]) {
```
Comment (nitaliano) [inline MapboxMobileEvents/MMEAPIClient.m]: @rclee should the base and SDK version be split into separate conditionals? On Android I won't be overriding the SDK version so it stays as the native version in the meantime.

**Comment by julianrex** [comment]
Are these overrides just for the telemetry lib? Would the user's expectation be that this would also override the user agent for requests from core?

**Comment by rclee** [comment]
@julianrex Yes, the expectation is to override core to track metrics coming from React Native.

**Review by nitaliano — APPROVED** [review]
👍

**Pull request merged**